python/apps/taiga/src/taiga/workflows/services/__init__.py

```diff
@@ -128,7 +129,60 @@ async def update_workflow_status(
     updated_status = await workflows_repositories.update_workflow_status(workflow_status=workflow_status, values=values)
 
     await workflows_events.emit_event_when_workflow_status_is_updated(
-        project=workflow.project, workflow_status=updated_status
+        project=workflow_status.project, workflow_status=updated_status
     )
 
     return updated_status
+
+
+##########################################################
+# delete workflow status
+##########################################################
+
+
+async def delete_workflow_status(workflow_status: WorkflowStatus, move_to_status_slug: str | None) -> bool:
+    """
+    This method deletes a workflow status, providing the option to first migrating its stories to another workflow
+    status of the same workflow.
+
+    :param workflow_status: the workflow status to delete
+    :param move_to_status_slug: the workflow status's slug to which move the stories from the status being deleted
+        - if not received, all the workflow status and its contained stories will be deleted
+        - if received, the workflow status will be deleted but its contained stories won't (they will be first moved to
+         the specified status)
+    :return: None
+    """
+    if move_to_status_slug:
+        move_to_status = await get_workflow_status(
+            project_id=workflow_status.project.id,
+            workflow_slug=workflow_status.workflow.slug,
+            status_slug=move_to_status_slug,
+        )
+        if not move_to_status:
+            raise ex.NonExistingMoveToStatus(f"The status '{move_to_status_slug}' doesn't exist")
+        if move_to_status == workflow_status:
+            raise ex.SameMoveToStatus("The stories can't be moved to the workflow status being deleted")
+
+        stories_to_move = await stories_repositories.list_stories(
+            filters={
+                "status_id": workflow_status.id,
+            },
+            order_by=["order"],
+        )
+
+        if stories_to_move:
+            await stories_services.reorder_stories(
+                project=workflow_status.project,
+                workflow=workflow_status.workflow,
+                target_status_slug=move_to_status_slug,
+                stories_refs=[story.ref for story in stories_to_move],
+            )
+
+    deleted = await workflows_repositories.delete_status(filters={"id": workflow_status.id})
+    if deleted > 0:
+        await workflows_events.emit_event_when_workflow_status_is_deleted(
+            project=workflow_status.project, workflow_status=workflow_status
+        )
+        return True
+
+    return False
```

Comment:
what does it mean here that there is no deletion?

Reply:
Yes, it's a typical response returned in the deletes methods, despite I don't make use of the returned value. 
I can change the method to avoid returning anything.

Reply:
no, no, you were right, in the services we use this form.

Reply:
The change should be now reverted to return a boolean